src/sensor/ping.cpp

```diff
@@ -552,6 +552,17 @@ void Ping::resetSensorLocalVariables()
     _lastPingConfigurationSrcId = -1;
 }
 
+QQuickItem* Ping::controlPanel(QObject *parent)
+{
+    QQmlEngine *engine = qmlEngine(parent);
+    if(!engine) {
+        qCDebug(PING_PROTOCOL_PING) << "No qml engine to load visualization.";
+        return nullptr;
+    }
+    QQmlComponent *component = new QQmlComponent(engine, QUrl("qrc:/Ping1DControlPanel.qml"));
```

Comment:
How is this deleted? Does engine own it and do the clean up?

Reply:
Added smart pointer to remove any possible memory leak, the parent should take of it, but better safe than sorry.